src/tools/assistant/listKnowledgeSources.ts

```diff
@@ -0,0 +1,74 @@
+import { CallToolResult } from "@modelcontextprotocol/sdk/types.js";
+import { formatUntrustedData, type OperationType } from "../tool.js";
+import { AssistantToolBase } from "./assistantTool.js";
+import { LogId } from "../../common/logger.js";
+import { stringify as yamlStringify } from "yaml";
+
+export type KnowledgeSource = {
+    /** The name of the data source */
+    id: string;
+    /** The type of the data source */
+    type: string;
+    /** A list of available versions for this data source */
+    versions: {
+        /** The version label of the data source */
+        label: string;
+        /** Whether this version is the current/default version */
+        isCurrent: boolean;
+    }[];
+};
+
+export type ListKnowledgeSourcesResponse = {
+    dataSources: KnowledgeSource[];
+};
+
+export const ListKnowledgeSourcesToolName = "list-knowledge-sources";
+
+export class ListKnowledgeSourcesTool extends AssistantToolBase {
+    public name = ListKnowledgeSourcesToolName;
+    protected description = "List available data sources in the MongoDB Assistant knowledge base";
```

Comment:
When testing locally I was able to trigger it explicitly by "listing knowledge topics.
After extending description a little bit I was able to call this tool more accurately.

"list mongodb topics for me"
```suggestion
    protected description = "List available data sources in the MongoDB Assistant knowledge base. Covers t knowledge topics for MongoDB database";
```